src/v/model/offset_interval.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Redpanda Data, Inc.
```

Comment:
When reader config is reused the bounds can be reset too. It would be done only once, when the reader is created. By once I mean that is done once per read "make reader" operation.